### ISSUE TITLE
MIR borrowck: the value of an assignment statement is uninitialized

### ISSUE THREAD

**Issue by arielb1** [rust-lang/rust#46159]
e.g..

```Rust
fn main() {
    let y;
    &(y = 0);
}
```

Causes this spurious error:
```
error[E0381]: borrow of possibly uninitialized variable: `_` (Mir)
 --> ack.rs:3:5
  |
3 |     &(y = 0);
  |     ^^^^^^^^ use of possibly uninitialized `_`

error: aborting due to previous error
```
